In [1]:
import pandas as pd

# **Load CMS Data**

In [2]:
from google.colab import files

uploaded = files.upload()

Saving FY_2026_Hospital_Readmissions_Reduction_Program_Hospital.csv to FY_2026_Hospital_Readmissions_Reduction_Program_Hospital.csv


In [3]:
uploaded.keys()

dict_keys(['FY_2026_Hospital_Readmissions_Reduction_Program_Hospital.csv'])

In [4]:
df = pd.read_csv("FY_2026_Hospital_Readmissions_Reduction_Program_Hospital.csv")

## **Inspect and Clean Data**

In [5]:
df.head()

,Facility Name,Facility ID,State,Measure Name,Number of Discharges,Footnote,Excess Readmission Ratio,Predicted Readmission Rate,Expected Readmission Rate,Number of Readmissions,Start Date,End Date
0,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-HIP-KNEE-HRRP,NaN,NaN,0.9875,4.5734,4.6311,Too Few to Report,07/01/2021,06/30/2024
1,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-CABG-HRRP,137.0,NaN,0.9531,10.3960,10.9078,13,07/01/2021,06/30/2024
2,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-AMI-HRRP,273.0,NaN,0.9370,13.2998,14.1948,33,07/01/2021,06/30/2024
3,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-COPD-HRRP,122.0,NaN,0.9823,16.6384,16.9389,19,07/01/2021,06/30/2024
4,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-PN-HRRP,507.0,NaN,0.9871,15.7529,15.9591,79,07/01/2021,06/30/2024


In [6]:
df.shape

(18330, 12)

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18330 entries, 0 to 18329
Data columns (total 12 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Facility Name               18330 non-null  object 
 1   Facility ID                 18330 non-null  int64  
 2   State                       18330 non-null  object 
 3   Measure Name                18330 non-null  object 
 4   Number of Discharges        8242 non-null   float64
 5   Footnote                    6987 non-null   float64
 6   Excess Readmission Ratio    11720 non-null  float64
 7   Predicted Readmission Rate  11720 non-null  float64
 8   Expected Readmission Rate   11720 non-null  float64
 9   Number of Readmissions      11720 non-null  object 
 10  Start Date                  18330 non-null  object 
 11  End Date                    18330 non-null  object 
dtypes: float64(5), int64(1), object(6)
memory usage: 1.7+ MB


In [8]:
df.isnull().sum()

,0
Facility Name,0
Facility ID,0
State,0
Measure Name,0
Number of Discharges,10088
Footnote,11343
Excess Readmission Ratio,6610
Predicted Readmission Rate,6610
Expected Readmission Rate,6610
Number of Readmissions,6610


In [9]:
df['Facility ID'].nunique(), df['Facility Name'].nunique()

(3055, 2995)

In [10]:
df['Measure Name'].value_counts()

,count
Measure Name,
READM-30-HIP-KNEE-HRRP,3055
READM-30-CABG-HRRP,3055
READM-30-AMI-HRRP,3055
READM-30-COPD-HRRP,3055
READM-30-PN-HRRP,3055
READM-30-HF-HRRP,3055


In [11]:
df['Number of Readmissions'].value_counts().head(15)

,count
Number of Readmissions,
Too Few to Report,3683
11,318
14,290
12,263
15,243
13,236
17,218
19,213
18,213


In [12]:
df["Number of Readmissions"].unique()[:20]

array(['Too Few to Report', '13', '33', '19', '79', '136', nan, '17',
       '35', '30', '29', '105', '88', '37', '16', '12', '11', '56', '46',
       '25'], dtype=object)

In [13]:
df['Footnote'].value_counts(dropna='False')

,count
Footnote,
5.0,3255
1.0,3150
29.0,377
7.0,205


In [14]:
missing = (
    df.isnull().sum()
      .to_frame("Missing Count")
)

missing["Missing %"] = (
    missing["Missing Count"] / len(df) * 100
).round(1)

missing

,Missing Count,Missing %
Facility Name,0,0.0
Facility ID,0,0.0
State,0,0.0
Measure Name,0,0.0
Number of Discharges,10088,55.0
Footnote,11343,61.9
Excess Readmission Ratio,6610,36.1
Predicted Readmission Rate,6610,36.1
Expected Readmission Rate,6610,36.1
Number of Readmissions,6610,36.1


In [15]:
df.groupby("Footnote")[
    [
        "Number of Discharges",
        "Excess Readmission Ratio",
        "Predicted Readmission Rate",
        "Expected Readmission Rate",
        "Number of Readmissions"
    ]
].count()

,Number of Discharges,Excess Readmission Ratio,Predicted Readmission Rate,Expected Readmission Rate,Number of Readmissions
Footnote,,,,,
1.0,0,0,0,0,0
5.0,0,0,0,0,0
7.0,205,0,0,0,0
29.0,239,377,377,377,377


In [16]:
pd.crosstab(
    df["Footnote"],
    df["Measure Name"],
    dropna=False
)

Measure Name,READM-30-AMI-HRRP,READM-30-CABG-HRRP,READM-30-COPD-HRRP,READM-30-HF-HRRP,READM-30-HIP-KNEE-HRRP,READM-30-PN-HRRP
Footnote,,,,,,
1.0,933,184,526,230,1114,163
5.0,319,1975,199,196,397,169
7.0,67,18,7,8,97,8
29.0,61,34,71,78,50,83
NaN,1675,844,2252,2543,1397,2632


In [17]:
df[df["Footnote"].notna()].head(20)

,Facility Name,Facility ID,State,Measure Name,Number of Discharges,Footnote,Excess Readmission Ratio,Predicted Readmission Rate,Expected Readmission Rate,Number of Readmissions,Start Date,End Date
6,MARSHALL MEDICAL CENTERS,10005,AL,READM-30-CABG-HRRP,NaN,5.0,NaN,NaN,NaN,NaN,07/01/2021,06/30/2024
7,MARSHALL MEDICAL CENTERS,10005,AL,READM-30-AMI-HRRP,NaN,1.0,NaN,NaN,NaN,NaN,07/01/2021,06/30/2024
19,MIZELL MEMORIAL HOSPITAL,10007,AL,READM-30-AMI-HRRP,NaN,1.0,NaN,NaN,NaN,NaN,07/01/2021,06/30/2024
21,MIZELL MEMORIAL HOSPITAL,10007,AL,READM-30-HIP-KNEE-HRRP,NaN,1.0,NaN,NaN,NaN,NaN,07/01/2021,06/30/2024
23,MIZELL MEMORIAL HOSPITAL,10007,AL,READM-30-CABG-HRRP,NaN,5.0,NaN,NaN,NaN,NaN,07/01/2021,06/30/2024
24,CRENSHAW COMMUNITY HOSPITAL,10008,AL,READM-30-HIP-KNEE-HRRP,NaN,5.0,NaN,NaN,NaN,NaN,07/01/2021,06/30/2024
26,CRENSHAW COMMUNITY HOSPITAL,10008,AL,READM-30-HF-HRRP,NaN,1.0,NaN,NaN,NaN,NaN,07/01/2021,06/30/2024
27,CRENSHAW COMMUNITY HOSPITAL,10008,AL,READM-30-CABG-HRRP,NaN,5.0,NaN,NaN,NaN,NaN,07/01/2021,06/30/2024
28,CRENSHAW COMMUNITY HOSPITAL,10008,AL,READM-30-COPD-HRRP,NaN,1.0,NaN,NaN,NaN,NaN,07/01/2021,06/30/2024
29,CRENSHAW COMMUNITY HOSPITAL,10008,AL,READM-30-AMI-HRRP,NaN,1.0,NaN,NaN,NaN,NaN,07/01/2021,06/30/2024


**Observations:**

1. The CMS HRRP dataset contains 3,055 facilities evaluated across six readmission measures.
2. Approximately 36% of facility-condition records do not contain calculated readmission performance metrics
3. the missingness appears to be associated with CMS reporting/eligibility footnotes rather than random data loss.
4. CMS explains footnote 29 as a case where the measure was calculated using partial performance-period data because of a CMS-approved exception.
5. Footnote 1 and 5 don't have any data for the readmission performance metrics
6. Footnote 7 has data for number of discharges but not for readmission performance metrics.

In [18]:
# Creating a Copy of the original data
df_analysis = df[
    df["Excess Readmission Ratio"].notna()
].copy()

In [19]:
df_analysis

,Facility Name,Facility ID,State,Measure Name,Number of Discharges,Footnote,Excess Readmission Ratio,Predicted Readmission Rate,Expected Readmission Rate,Number of Readmissions,Start Date,End Date
0,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-HIP-KNEE-HRRP,NaN,NaN,0.9875,4.5734,4.6311,Too Few to Report,07/01/2021,06/30/2024
1,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-CABG-HRRP,137.0,NaN,0.9531,10.3960,10.9078,13,07/01/2021,06/30/2024
2,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-AMI-HRRP,273.0,NaN,0.9370,13.2998,14.1948,33,07/01/2021,06/30/2024
3,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-COPD-HRRP,122.0,NaN,0.9823,16.6384,16.9389,19,07/01/2021,06/30/2024
4,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-PN-HRRP,507.0,NaN,0.9871,15.7529,15.9591,79,07/01/2021,06/30/2024
...,...,...,...,...,...,...,...,...,...,...,...,...
18246,METHODIST MIDLOTHIAN MEDICAL CENTER,670300,TX,READM-30-PN-HRRP,NaN,NaN,0.9886,13.1557,13.3070,Too Few to Report,07/01/2021,06/30/2024
18248,METHODIST MIDLOTHIAN MEDICAL CENTER,670300,TX,READM-30-HF-HRRP,NaN,NaN,0.9782,16.7151,17.0871,Too Few to Report,07/01/2021,06/30/2024
18249,METHODIST MIDLOTHIAN MEDICAL CENTER,670300,TX,READM-30-COPD-HRRP,NaN,NaN,0.9917,17.0178,17.1609,Too Few to Report,07/01/2021,06/30/2024
18252,TEXAS HEALTH HOSPITAL MANSFIELD,670309,TX,READM-30-PN-HRRP,NaN,NaN,1.0080,14.9268,14.8077,Too Few to Report,07/01/2021,06/30/2024


In [20]:
df_analysis["Partial_Period_Flag"] = (df_analysis["Footnote"] == 29)

#377 performance observations were based on partial-period reporting and were retained but flagged for interpretation.

In [21]:
df_analysis.head()

,Facility Name,Facility ID,State,Measure Name,Number of Discharges,Footnote,Excess Readmission Ratio,Predicted Readmission Rate,Expected Readmission Rate,Number of Readmissions,Start Date,End Date,Partial_Period_Flag
0,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-HIP-KNEE-HRRP,NaN,NaN,0.9875,4.5734,4.6311,Too Few to Report,07/01/2021,06/30/2024,False
1,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-CABG-HRRP,137.0,NaN,0.9531,10.3960,10.9078,13,07/01/2021,06/30/2024,False
2,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-AMI-HRRP,273.0,NaN,0.9370,13.2998,14.1948,33,07/01/2021,06/30/2024,False
3,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-COPD-HRRP,122.0,NaN,0.9823,16.6384,16.9389,19,07/01/2021,06/30/2024,False
4,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-PN-HRRP,507.0,NaN,0.9871,15.7529,15.9591,79,07/01/2021,06/30/2024,False


In [22]:
df_analysis["Number of Readmissions Numeric"] = pd.to_numeric(df_analysis["Number of Readmissions"],errors="coerce")

In [23]:
df_analysis.head()

,Facility Name,Facility ID,State,Measure Name,Number of Discharges,Footnote,Excess Readmission Ratio,Predicted Readmission Rate,Expected Readmission Rate,Number of Readmissions,Start Date,End Date,Partial_Period_Flag,Number of Readmissions Numeric
0,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-HIP-KNEE-HRRP,NaN,NaN,0.9875,4.5734,4.6311,Too Few to Report,07/01/2021,06/30/2024,False,NaN
1,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-CABG-HRRP,137.0,NaN,0.9531,10.3960,10.9078,13,07/01/2021,06/30/2024,False,13.0
2,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-AMI-HRRP,273.0,NaN,0.9370,13.2998,14.1948,33,07/01/2021,06/30/2024,False,33.0
3,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-COPD-HRRP,122.0,NaN,0.9823,16.6384,16.9389,19,07/01/2021,06/30/2024,False,19.0
4,SOUTHEAST HEALTH MEDICAL CENTER,10001,AL,READM-30-PN-HRRP,507.0,NaN,0.9871,15.7529,15.9591,79,07/01/2021,06/30/2024,False,79.0


In [24]:
df_analysis.shape

(11720, 14)

In [25]:
df_analysis.info()

<class 'pandas.core.frame.DataFrame'>
Index: 11720 entries, 0 to 18254
Data columns (total 14 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Facility Name                   11720 non-null  object 
 1   Facility ID                     11720 non-null  int64  
 2   State                           11720 non-null  object 
 3   Measure Name                    11720 non-null  object 
 4   Number of Discharges            8037 non-null   float64
 5   Footnote                        377 non-null    float64
 6   Excess Readmission Ratio        11720 non-null  float64
 7   Predicted Readmission Rate      11720 non-null  float64
 8   Expected Readmission Rate       11720 non-null  float64
 9   Number of Readmissions          11720 non-null  object 
 10  Start Date                      11720 non-null  object 
 11  End Date                        11720 non-null  object 
 12  Partial_Period_Flag             11720

In [26]:
df_analysis[
    [
        "Excess Readmission Ratio",
        "Predicted Readmission Rate",
        "Expected Readmission Rate",
        "Number of Readmissions Numeric",
        "Partial_Period_Flag"
    ]
].isnull().sum()

,0
Excess Readmission Ratio,0
Predicted Readmission Rate,0
Expected Readmission Rate,0
Number of Readmissions Numeric,3683
Partial_Period_Flag,0


# **Exploratory Data Analysis**

In [27]:
#Q1: Which readmission condition has the highest average excess readmission ratio
condition_summary = df_analysis.groupby('Measure Name').agg(
    Avg_Excess_Readmission_Ratio = ('Excess Readmission Ratio','mean'),
    Avg_Predicted_Readmission_Rate = ('Predicted Readmission Rate','mean'),
    Avg_Expected_Readmission_Rate = ('Expected Readmission Rate','mean'),
    Hospitals=("Facility ID", "nunique")
).sort_values('Avg_Excess_Readmission_Ratio', ascending=False)

condition_summary

,Avg_Excess_Readmission_Ratio,Avg_Predicted_Readmission_Rate,Avg_Expected_Readmission_Rate,Hospitals
Measure Name,,,,
READM-30-HIP-KNEE-HRRP,1.003985,5.491330,5.456239,1447
READM-30-CABG-HRRP,1.001789,10.675471,10.650657,878
READM-30-AMI-HRRP,1.001755,13.421979,13.384528,1736
READM-30-PN-HRRP,1.001486,15.597373,15.558561,2715
READM-30-HF-HRRP,1.001439,19.381564,19.347008,2621
READM-30-COPD-HRRP,1.001133,18.001379,17.976027,2323


Interpretation:
1. Across hospitals with reported data, hip/knee replacement had the highest average Excess Readmission Ratio, with predicted readmissions about 0.4% above the expected benchmark on average.
2. Heart Failure has the highest average predicted readmission rate, but its excess readmission ratio remains close to 1 because its expected rate is similarly high
3. CABG had one of the higher average Excess Readmission Ratios, but the result was based on fewer reporting hospitals than the other measures, so comparisons are interpreted with some caution.


In [28]:
#Q2: Which state has the highest average excess readmission ratio
state_summary = df_analysis.groupby('State').agg(
    Avg_Excess_Readmission_Ratio = ('Excess Readmission Ratio','mean'),
    Avg_Predicted_Readmission_Rate = ('Predicted Readmission Rate','mean'),
    Avg_Expected_Readmission_Rate = ('Expected Readmission Rate','mean'),
    Hospitals=("Facility ID", "nunique")
).sort_values('Avg_Excess_Readmission_Ratio', ascending=False)

state_summary.head(10)

,Avg_Excess_Readmission_Ratio,Avg_Predicted_Readmission_Rate,Avg_Expected_Readmission_Rate,Hospitals
State,,,,
MA,1.034428,14.908686,14.383138,52
NJ,1.027668,16.036789,15.615087,61
FL,1.022866,15.783541,15.467182,164
IL,1.019390,15.784370,15.529866,109
MS,1.014094,15.185147,14.945437,50
AL,1.013884,14.662347,14.509635,67
WV,1.011502,15.952066,15.695644,22
NV,1.011438,14.803573,14.587387,19
GA,1.011403,15.464458,15.308561,88


Interpretation:
1. Massachusetts had the highest average Excess Readmission Ratio among states in this analysis, with predicted readmissions approximately 3.4% above expected levels on average.

In [29]:
#Q3: Which condition in MA state has the highest average excess readmission ratio
state_condition_summary = (
    df_analysis
    .groupby(["State", "Measure Name"])
    .agg(
        Avg_Excess_Readmission_Ratio=(
            "Excess Readmission Ratio", "mean"
        ),
        Avg_Predicted_Readmission_Rate=(
            "Predicted Readmission Rate", "mean"
        ),
        Avg_Expected_Readmission_Rate=(
            "Expected Readmission Rate", "mean"
        ),
        Hospitals=(
            "Facility ID", "nunique"
        )
    )
    .reset_index()
)

state_condition_summary[
    state_condition_summary["State"] == "MA"
].sort_values(
    "Avg_Excess_Readmission_Ratio",
    ascending=False
)

,State,Measure Name,Avg_Excess_Readmission_Ratio,Avg_Predicted_Readmission_Rate,Avg_Expected_Readmission_Rate,Hospitals
114,MA,READM-30-AMI-HRRP,1.051164,14.472061,13.761903,33
119,MA,READM-30-PN-HRRP,1.045849,16.111329,15.419755,51
117,MA,READM-30-HF-HRRP,1.036067,19.854269,19.195906,51
116,MA,READM-30-COPD-HRRP,1.035729,18.520488,17.905669,49
115,MA,READM-30-CABG-HRRP,1.024115,10.906508,10.653454,13
118,MA,READM-30-HIP-KNEE-HRRP,1.007730,5.045874,5.035786,43


Interpretation:
Massachusetts had the highest average state-level Excess Readmission Ratio in the analyzed CMS data. AMI showed the largest relative gap, with predicted readmissions about 5.1% above expected levels, while several other conditions also exceeded their expected benchmarks, suggesting the state-level result was broad-based rather than attributable to a single condition.

In [30]:
#Q4: Which hospitals are associated with high excess readmission ratio in MA for AMI

ma_ami = (
    df_analysis[
        (df_analysis["State"] == "MA") &
        (df_analysis["Measure Name"] == "READM-30-AMI-HRRP")
    ]
    .sort_values(
        "Excess Readmission Ratio",
        ascending=False
    )
)

ma_ami[
    [
        "Facility ID",
        "Facility Name",
        "Excess Readmission Ratio",
        "Predicted Readmission Rate",
        "Expected Readmission Rate",
        "Number of Readmissions Numeric"
    ]
].head(10)

,Facility ID,Facility Name,Excess Readmission Ratio,Predicted Readmission Rate,Expected Readmission Rate,Number of Readmissions Numeric
8003,220036,BOSTON MEDICAL CENTER-BRIGHTON,1.2639,18.4135,14.5689,73.0
8165,220110,BRIGHAM AND WOMEN'S HOSPITAL,1.2243,17.3296,14.1543,125.0
8177,220116,TUFTS MEDICAL CENTER,1.2234,18.8957,15.4454,48.0
8141,220100,SOUTH SHORE HOSPITAL,1.1911,15.9296,13.3736,65.0
8087,220077,BAYSTATE MEDICAL CENTER,1.1792,15.4378,13.0921,182.0
8111,220086,BETH ISRAEL DEACONESS MEDICAL CENTER,1.1194,15.5018,13.8489,88.0
8039,220063,LOWELL GENERAL HOSPITAL,1.1177,15.5400,13.9040,39.0
7931,220012,CAPE COD HOSPITAL,1.0987,12.4680,11.3481,86.0
7919,220010,LAWRENCE GENERAL HOSPITAL,1.0935,14.2758,13.0548,20.0
8153,220105,WINCHESTER HOSPITAL,1.0790,16.1460,14.9637,NaN


In [31]:
ma_ami["Excess Readmission Ratio"].describe()

,Excess Readmission Ratio
count,33.000000
mean,1.051164
std,0.087561
min,0.864600
25%,0.998200
50%,1.028500
75%,1.093500
max,1.263900


In [32]:
(ma_ami["Excess Readmission Ratio"] > 1).value_counts()

,count
Excess Readmission Ratio,
True,23
False,10


In [33]:
above_expected = (
    ma_ami["Excess Readmission Ratio"] > 1
).mean() * 100

above_expected

np.float64(69.6969696969697)

Interpretation:
1. So Boston Medical Center–Brighton’s predicted AMI readmission rate is about 26.4% higher than its expected benchmark.
2. Massachusetts’ elevated AMI readmission performance appears to be broad-based: approximately 70% of reporting MA hospitals with AMI had ERR above expected benchmark, and the median ERR was 1.029. While several hospitals showed notably high ratios, the statewide pattern was not driven solely by a small number of outliers.
3. Hospitals with the highest absolute readmission rates or readmission counts are not necessarily the hospitals with the highest excess readmission ratios, because ERR evaluates performance relative to each hospital's risk-adjusted expected rate.

# **Load CMS Data - Additional Hospital Information**

In [53]:
from google.colab import files

uploaded = files.upload()

Saving Hospital_General_Information.csv to Hospital_General_Information.csv


In [54]:
df2  = pd.read_csv("Hospital_General_Information.csv")

In [55]:
df2.head()

,Facility ID,Facility Name,Address,City/Town,State,ZIP Code,County/Parish,Telephone Number,Hospital Type,Hospital Ownership,...,Count of READM Measures Better,Count of READM Measures No Different,Count of READM Measures Worse,READM Group Footnote,Pt Exp Group Measure Count,Count of Facility Pt Exp Measures,Pt Exp Group Footnote,TE Group Measure Count,Count of Facility TE Measures,TE Group Footnote
0,010001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,Acute Care Hospitals,Government - Hospital District or Authority,...,1,9,1,NaN,15,15,NaN,10,10,NaN
1,010005,MARSHALL MEDICAL CENTERS,2505 U S HIGHWAY 431 NORTH,BOAZ,AL,35957,MARSHALL,(256) 593-8310,Acute Care Hospitals,Government - Hospital District or Authority,...,1,8,0,NaN,15,15,NaN,10,10,NaN
2,010006,NORTH ALABAMA MEDICAL CENTER,1701 VETERANS DRIVE,FLORENCE,AL,35630,LAUDERDALE,(256) 768-8400,Acute Care Hospitals,Proprietary,...,1,8,0,NaN,15,15,NaN,10,9,NaN
3,010007,MIZELL MEMORIAL HOSPITAL,702 N MAIN ST,OPP,AL,36467,COVINGTON,(334) 493-3541,Acute Care Hospitals,Voluntary non-profit - Private,...,0,3,2,NaN,15,5,NaN,10,7,NaN
4,010011,ST. VINCENT'S EAST,50 MEDICAL PARK EAST DRIVE,BIRMINGHAM,AL,35235,JEFFERSON,(205) 838-3122,Acute Care Hospitals,Voluntary non-profit - Private,...,1,5,2,29.0,15,10,29.0,10,7,29.0


In [65]:
df2.shape

(5419, 38)

In [57]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5419 entries, 0 to 5418
Data columns (total 38 columns):
 #   Column                                            Non-Null Count  Dtype  
---  ------                                            --------------  -----  
 0   Facility ID                                       5419 non-null   object 
 1   Facility Name                                     5419 non-null   object 
 2   Address                                           5419 non-null   object 
 3   City/Town                                         5419 non-null   object 
 4   State                                             5419 non-null   object 
 5   ZIP Code                                          5419 non-null   int64  
 6   County/Parish                                     5419 non-null   object 
 7   Telephone Number                                  5419 non-null   object 
 8   Hospital Type                                     5419 non-null   object 
 9   Hospital Ownership 

In [64]:
df2.columns

Index(['Facility ID', 'Facility Name', 'Address', 'City/Town', 'State',
       'ZIP Code', 'County/Parish', 'Telephone Number', 'Hospital Type',
       'Hospital Ownership', 'Emergency Services',
       'Meets criteria for birthing friendly designation',
       'Hospital overall rating', 'Hospital overall rating footnote',
       'MORT Group Measure Count', 'Count of Facility MORT Measures',
       'Count of MORT Measures Better', 'Count of MORT Measures No Different',
       'Count of MORT Measures Worse', 'MORT Group Footnote',
       'Safety Group Measure Count', 'Count of Facility Safety Measures',
       'Count of Safety Measures Better',
       'Count of Safety Measures No Different',
       'Count of Safety Measures Worse', 'Safety Group Footnote',
       'READM Group Measure Count', 'Count of Facility READM Measures',
       'Count of READM Measures Better',
       'Count of READM Measures No Different', 'Count of READM Measures Worse',
       'READM Group Footnote', 'Pt Exp Gr

In [58]:
df2.isna().sum()

,0
Facility ID,0
Facility Name,0
Address,0
City/Town,0
State,0
ZIP Code,0
County/Parish,0
Telephone Number,0
Hospital Type,0
Hospital Ownership,0


In [60]:
df2['Facility ID'].nunique(), df2['Facility Name'].nunique()

(5419, 5288)

In [61]:
df2['Facility ID'].duplicated().sum()

np.int64(0)

In [66]:
# standardize the information in both dataframes

df_analysis["Facility ID"] = (
    df_analysis["Facility ID"]
    .astype(str)
    .str.zfill(6)
)

df2["Facility ID"] = (
    df2["Facility ID"]
    .astype(str)
    .str.strip()
    .str.zfill(6)
)

In [68]:
df_analysis["Facility ID"].head()

,Facility ID
0,010001
1,010001
2,010001
3,010001
4,010001


In [67]:
df2["Facility ID"].head()

,Facility ID
0,010001
1,010005
2,010006
3,010007
4,010011


In [69]:
# Keeping selected hospital info for analysis
hospital_attributes = df2[
    [
        "Facility ID",
        "Hospital Type",
        "Hospital Ownership",
        "Emergency Services",
        "Hospital overall rating"
    ]
].copy()

In [70]:
hospital_attributes

,Facility ID,Hospital Type,Hospital Ownership,Emergency Services,Hospital overall rating
0,010001,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4
1,010005,Acute Care Hospitals,Government - Hospital District or Authority,Yes,3
2,010006,Acute Care Hospitals,Proprietary,Yes,2
3,010007,Acute Care Hospitals,Voluntary non-profit - Private,Yes,1
4,010011,Acute Care Hospitals,Voluntary non-profit - Private,Yes,3
...,...,...,...,...,...
5414,671300,Critical Access Hospitals,Government - Hospital District or Authority,Yes,Not Available
5415,671301,Critical Access Hospitals,Government - Hospital District or Authority,Yes,Not Available
5416,671302,Critical Access Hospitals,Government - Hospital District or Authority,Yes,Not Available
5417,671303,Critical Access Hospitals,Government - Hospital District or Authority,Yes,Not Available


# **Merging the Two Datasets**

In [71]:
# Merging the two datasets
df_combined = df_analysis.merge(
    hospital_attributes,
    on="Facility ID",
    how="left",
    validate="many_to_one",
    indicator=True
)

In [73]:
df_combined.head()

,Facility Name,Facility ID,State,Measure Name,Number of Discharges,Footnote,Excess Readmission Ratio,Predicted Readmission Rate,Expected Readmission Rate,Number of Readmissions,Start Date,End Date,Partial_Period_Flag,Number of Readmissions Numeric,Hospital Type,Hospital Ownership,Emergency Services,Hospital overall rating,_merge
0,SOUTHEAST HEALTH MEDICAL CENTER,010001,AL,READM-30-HIP-KNEE-HRRP,NaN,NaN,0.9875,4.5734,4.6311,Too Few to Report,07/01/2021,06/30/2024,False,NaN,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4,both
1,SOUTHEAST HEALTH MEDICAL CENTER,010001,AL,READM-30-CABG-HRRP,137.0,NaN,0.9531,10.3960,10.9078,13,07/01/2021,06/30/2024,False,13.0,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4,both
2,SOUTHEAST HEALTH MEDICAL CENTER,010001,AL,READM-30-AMI-HRRP,273.0,NaN,0.9370,13.2998,14.1948,33,07/01/2021,06/30/2024,False,33.0,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4,both
3,SOUTHEAST HEALTH MEDICAL CENTER,010001,AL,READM-30-COPD-HRRP,122.0,NaN,0.9823,16.6384,16.9389,19,07/01/2021,06/30/2024,False,19.0,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4,both
4,SOUTHEAST HEALTH MEDICAL CENTER,010001,AL,READM-30-PN-HRRP,507.0,NaN,0.9871,15.7529,15.9591,79,07/01/2021,06/30/2024,False,79.0,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4,both


In [74]:
df_analysis.shape, df_combined.shape

((11720, 14), (11720, 19))

In [76]:
df_combined["_merge"].value_counts()

,count
_merge,
both,11681
left_only,39
right_only,0


In [77]:
df_combined[
    [
        "Facility ID",
        "Facility Name",
        "Measure Name",
        "Excess Readmission Ratio",
        "Hospital Type",
        "Hospital Ownership",
        "Emergency Services",
        "Hospital overall rating"
    ]
].head(10)

,Facility ID,Facility Name,Measure Name,Excess Readmission Ratio,Hospital Type,Hospital Ownership,Emergency Services,Hospital overall rating
0,010001,SOUTHEAST HEALTH MEDICAL CENTER,READM-30-HIP-KNEE-HRRP,0.9875,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4
1,010001,SOUTHEAST HEALTH MEDICAL CENTER,READM-30-CABG-HRRP,0.9531,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4
2,010001,SOUTHEAST HEALTH MEDICAL CENTER,READM-30-AMI-HRRP,0.9370,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4
3,010001,SOUTHEAST HEALTH MEDICAL CENTER,READM-30-COPD-HRRP,0.9823,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4
4,010001,SOUTHEAST HEALTH MEDICAL CENTER,READM-30-PN-HRRP,0.9871,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4
5,010001,SOUTHEAST HEALTH MEDICAL CENTER,READM-30-HF-HRRP,1.0233,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4
6,010005,MARSHALL MEDICAL CENTERS,READM-30-COPD-HRRP,0.9308,Acute Care Hospitals,Government - Hospital District or Authority,Yes,3
7,010005,MARSHALL MEDICAL CENTERS,READM-30-HF-HRRP,1.0087,Acute Care Hospitals,Government - Hospital District or Authority,Yes,3
8,010005,MARSHALL MEDICAL CENTERS,READM-30-HIP-KNEE-HRRP,0.8602,Acute Care Hospitals,Government - Hospital District or Authority,Yes,3
9,010005,MARSHALL MEDICAL CENTERS,READM-30-PN-HRRP,0.8727,Acute Care Hospitals,Government - Hospital District or Authority,Yes,3


In [78]:
unmatched = df_combined[
    df_combined["_merge"] == "left_only"
]

unmatched[
    [
        "Facility ID",
        "Facility Name",
        "State",
        "Measure Name"
    ]
].drop_duplicates()

,Facility ID,Facility Name,State,Measure Name
19,010008,CRENSHAW COMMUNITY HOSPITAL,AL,READM-30-PN-HRRP
114,010059,LAWRENCE MEDICAL CENTER,AL,READM-30-PN-HRRP
202,010120,MONROE COUNTY HOSPITAL,AL,READM-30-HF-HRRP
203,010120,MONROE COUNTY HOSPITAL,AL,READM-30-PN-HRRP
609,040067,MAGNOLIA REGIONAL MEDICAL HOSPITAL,AR,READM-30-HF-HRRP
610,040067,MAGNOLIA REGIONAL MEDICAL HOSPITAL,AR,READM-30-PN-HRRP
938,050125,REGIONAL MEDICAL CENTER OF SAN JOSE,CA,READM-30-HF-HRRP
939,050125,REGIONAL MEDICAL CENTER OF SAN JOSE,CA,READM-30-AMI-HRRP
940,050125,REGIONAL MEDICAL CENTER OF SAN JOSE,CA,READM-30-HIP-KNEE-HRRP
941,050125,REGIONAL MEDICAL CENTER OF SAN JOSE,CA,READM-30-COPD-HRRP


In [79]:
unmatched["Facility ID"].nunique()

15

In [80]:
df_hospital_analysis = df_combined[
    df_combined["_merge"] == "both"
].copy()

In [81]:
df_hospital_analysis.head()

,Facility Name,Facility ID,State,Measure Name,Number of Discharges,Footnote,Excess Readmission Ratio,Predicted Readmission Rate,Expected Readmission Rate,Number of Readmissions,Start Date,End Date,Partial_Period_Flag,Number of Readmissions Numeric,Hospital Type,Hospital Ownership,Emergency Services,Hospital overall rating,_merge
0,SOUTHEAST HEALTH MEDICAL CENTER,010001,AL,READM-30-HIP-KNEE-HRRP,NaN,NaN,0.9875,4.5734,4.6311,Too Few to Report,07/01/2021,06/30/2024,False,NaN,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4,both
1,SOUTHEAST HEALTH MEDICAL CENTER,010001,AL,READM-30-CABG-HRRP,137.0,NaN,0.9531,10.3960,10.9078,13,07/01/2021,06/30/2024,False,13.0,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4,both
2,SOUTHEAST HEALTH MEDICAL CENTER,010001,AL,READM-30-AMI-HRRP,273.0,NaN,0.9370,13.2998,14.1948,33,07/01/2021,06/30/2024,False,33.0,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4,both
3,SOUTHEAST HEALTH MEDICAL CENTER,010001,AL,READM-30-COPD-HRRP,122.0,NaN,0.9823,16.6384,16.9389,19,07/01/2021,06/30/2024,False,19.0,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4,both
4,SOUTHEAST HEALTH MEDICAL CENTER,010001,AL,READM-30-PN-HRRP,507.0,NaN,0.9871,15.7529,15.9591,79,07/01/2021,06/30/2024,False,79.0,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4,both


In [82]:
df_hospital_analysis.drop(
    columns=["_merge","Start Date","End Date"],
    inplace=True
)

In [83]:
df_hospital_analysis

,Facility Name,Facility ID,State,Measure Name,Number of Discharges,Footnote,Excess Readmission Ratio,Predicted Readmission Rate,Expected Readmission Rate,Number of Readmissions,Partial_Period_Flag,Number of Readmissions Numeric,Hospital Type,Hospital Ownership,Emergency Services,Hospital overall rating
0,SOUTHEAST HEALTH MEDICAL CENTER,010001,AL,READM-30-HIP-KNEE-HRRP,NaN,NaN,0.9875,4.5734,4.6311,Too Few to Report,False,NaN,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4
1,SOUTHEAST HEALTH MEDICAL CENTER,010001,AL,READM-30-CABG-HRRP,137.0,NaN,0.9531,10.3960,10.9078,13,False,13.0,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4
2,SOUTHEAST HEALTH MEDICAL CENTER,010001,AL,READM-30-AMI-HRRP,273.0,NaN,0.9370,13.2998,14.1948,33,False,33.0,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4
3,SOUTHEAST HEALTH MEDICAL CENTER,010001,AL,READM-30-COPD-HRRP,122.0,NaN,0.9823,16.6384,16.9389,19,False,19.0,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4
4,SOUTHEAST HEALTH MEDICAL CENTER,010001,AL,READM-30-PN-HRRP,507.0,NaN,0.9871,15.7529,15.9591,79,False,79.0,Acute Care Hospitals,Government - Hospital District or Authority,Yes,4
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
11715,METHODIST MIDLOTHIAN MEDICAL CENTER,670300,TX,READM-30-PN-HRRP,NaN,NaN,0.9886,13.1557,13.3070,Too Few to Report,False,NaN,Acute Care Hospitals,Voluntary non-profit - Private,Yes,4
11716,METHODIST MIDLOTHIAN MEDICAL CENTER,670300,TX,READM-30-HF-HRRP,NaN,NaN,0.9782,16.7151,17.0871,Too Few to Report,False,NaN,Acute Care Hospitals,Voluntary non-profit - Private,Yes,4
11717,METHODIST MIDLOTHIAN MEDICAL CENTER,670300,TX,READM-30-COPD-HRRP,NaN,NaN,0.9917,17.0178,17.1609,Too Few to Report,False,NaN,Acute Care Hospitals,Voluntary non-profit - Private,Yes,4
11718,TEXAS HEALTH HOSPITAL MANSFIELD,670309,TX,READM-30-PN-HRRP,NaN,NaN,1.0080,14.9268,14.8077,Too Few to Report,False,NaN,Acute Care Hospitals,Voluntary non-profit - Private,Yes,5


# **Analysing additional information**

In [89]:
# Do average Excess Readmission Ratios differ by hospital ownership type?
hospital_level_err = (
    df_hospital_analysis
    .groupby(
        ["Facility ID", "Hospital Ownership"],
        as_index=False
    )
    .agg(
        Hospital_Avg_ERR=(
            "Excess Readmission Ratio",
            "mean"
        )
    )
)

In [91]:
ownership_summary = (
    hospital_level_err
    .groupby("Hospital Ownership")
    .agg(
        Avg_ERR=(
            "Hospital_Avg_ERR",
            "mean"
        ),
        Median_ERR=(
            "Hospital_Avg_ERR",
            "median"
        ),
        Hospitals=(
            "Facility ID",
            "nunique"
        )
    )
    .sort_values(
        "Avg_ERR",
        ascending=False
    ))

In [92]:
ownership_summary

,Avg_ERR,Median_ERR,Hospitals
Hospital Ownership,,,
Proprietary,1.012651,1.011725,544
Government - Federal,1.005336,1.015500,13
Government - State,1.003910,1.002167,37
Government - Local,1.002423,1.001150,129
Government - Hospital District or Authority,0.999061,0.996867,206
Voluntary non-profit - Private,0.998632,0.997362,1404
Voluntary non-profit - Other,0.996176,0.994200,229
Voluntary non-profit - Church,0.991822,0.987317,195
Tribal,0.975807,0.963800,5


Interpretation: Hospital-level average ERR varied modestly by ownership type. Proprietary hospitals had the highest average among the larger ownership categories, while several nonprofit groups were slightly below the expected benchmark.


In [93]:
# Does this ownership pattern still appear when we compare hospitals within the same clinical condition?
ownership_condition = (
    df_hospital_analysis
    .groupby(
        ["Hospital Ownership", "Measure Name"]
    )
    .agg(
        Avg_ERR=(
            "Excess Readmission Ratio",
            "mean"
        ),
        Hospitals=(
            "Facility ID",
            "nunique"
        )
    )
    .reset_index()
)

ownership_condition

,Hospital Ownership,Measure Name,Avg_ERR,Hospitals
0,Government - Federal,READM-30-AMI-HRRP,1.006225,4
1,Government - Federal,READM-30-CABG-HRRP,0.966100,2
2,Government - Federal,READM-30-COPD-HRRP,1.019043,7
3,Government - Federal,READM-30-HF-HRRP,1.046650,10
4,Government - Federal,READM-30-HIP-KNEE-HRRP,0.838333,3
5,Government - Federal,READM-30-PN-HRRP,1.003954,13
6,Government - Hospital District or Authority,READM-30-AMI-HRRP,1.001060,98
7,Government - Hospital District or Authority,READM-30-CABG-HRRP,1.002555,51
8,Government - Hospital District or Authority,READM-30-COPD-HRRP,0.998171,154
9,Government - Hospital District or Authority,READM-30-HF-HRRP,1.008805,180


In [94]:
ownership_condition_pivot = (
    ownership_condition
    .pivot(
        index="Hospital Ownership",
        columns="Measure Name",
        values="Avg_ERR"
    )
)

ownership_condition_pivot

Measure Name,READM-30-AMI-HRRP,READM-30-CABG-HRRP,READM-30-COPD-HRRP,READM-30-HF-HRRP,READM-30-HIP-KNEE-HRRP,READM-30-PN-HRRP
Hospital Ownership,,,,,,
Government - Federal,1.006225,0.966100,1.019043,1.046650,0.838333,1.003954
Government - Hospital District or Authority,1.001060,1.002555,0.998171,1.008805,0.996253,0.993113
Government - Local,0.999935,1.005347,1.002213,1.009062,1.002825,0.997371
Government - State,1.011522,1.005872,0.996506,0.997433,1.018200,0.988725
Physician,0.997255,0.975257,0.997462,0.991762,0.912989,0.991440
Proprietary,1.017254,1.034977,1.005495,1.015477,1.028097,1.016903
Tribal,NaN,NaN,1.013800,0.987975,NaN,0.958000
Voluntary non-profit - Church,0.996586,0.991591,0.992701,0.991232,0.999895,0.994645
Voluntary non-profit - Other,0.994360,1.006542,0.999065,1.000403,0.997841,0.997992


Interpretation: Proprietary hospitals showed average Excess Readmission Ratios above the expected benchmark across all six HRRP conditions in this dataset. The largest relative gaps were observed for CABG and hip/knee replacement. This indicates a consistent association between proprietary ownership and higher average ERR across conditions, but it does not establish that ownership causes higher readmissions.

In [95]:
ownership_condition_count = (
    ownership_condition
    .pivot(
        index="Hospital Ownership",
        columns="Measure Name",
        values="Hospitals"
    )
)

ownership_condition_count

Measure Name,READM-30-AMI-HRRP,READM-30-CABG-HRRP,READM-30-COPD-HRRP,READM-30-HF-HRRP,READM-30-HIP-KNEE-HRRP,READM-30-PN-HRRP
Hospital Ownership,,,,,,
Government - Federal,4.0,2.0,7.0,10.0,3.0,13.0
Government - Hospital District or Authority,98.0,51.0,154.0,180.0,81.0,203.0
Government - Local,52.0,17.0,100.0,122.0,51.0,126.0
Government - State,27.0,18.0,32.0,36.0,18.0,36.0
Physician,11.0,7.0,8.0,13.0,46.0,10.0
Proprietary,306.0,136.0,407.0,474.0,268.0,502.0
Tribal,NaN,NaN,1.0,4.0,NaN,5.0
Voluntary non-profit - Church,145.0,78.0,179.0,191.0,110.0,193.0
Voluntary non-profit - Other,136.0,55.0,198.0,223.0,113.0,224.0


Interpretation: Proprietary hospitals showed average ERR above 1.0 across all six HRRP measures, with each condition represented by more than 100 reporting hospitals. The association was therefore broad-based across conditions rather than driven by a single measure or very small sample.

In [96]:
# Do hospitals with higher overall ratings also tend to have lower Excess Readmission Ratios?
df_hospital_analysis["Hospital overall rating"].value_counts(dropna=False)

,count
Hospital overall rating,
3,3655
4,3343
2,2501
5,1274
1,663
Not Available,245


In [97]:
df_hospital_analysis["Hospital Rating Numeric"] = pd.to_numeric(
    df_hospital_analysis["Hospital overall rating"],
    errors="coerce"
)

In [98]:
df_hospital_analysis[
    ["Hospital overall rating", "Hospital Rating Numeric"]
].drop_duplicates()

,Hospital overall rating,Hospital Rating Numeric
0,4,4.0
6,3,3.0
10,2,2.0
16,1,1.0
40,Not Available,NaN
272,5,5.0


In [99]:
(
    df_hospital_analysis[
        ["Facility ID", "Hospital Rating Numeric"]
    ]
    .drop_duplicates()
    ["Hospital Rating Numeric"]
    .value_counts()
    .sort_index()
)

,count
Hospital Rating Numeric,
1.0,169
2.0,567
3.0,840
4.0,758
5.0,287


In [100]:
hospital_rating_err = (
    df_hospital_analysis
    .dropna(subset=["Hospital Rating Numeric"])
    .groupby(
        ["Facility ID", "Hospital Rating Numeric"],
        as_index=False
    )
    .agg(
        Hospital_Avg_ERR=(
            "Excess Readmission Ratio",
            "mean"
        )
    )
)

In [101]:
rating_summary = (
    hospital_rating_err
    .groupby("Hospital Rating Numeric")
    .agg(
        Avg_ERR=("Hospital_Avg_ERR", "mean"),
        Median_ERR=("Hospital_Avg_ERR", "median"),
        Hospitals=("Facility ID", "nunique")
    )
)

rating_summary

,Avg_ERR,Median_ERR,Hospitals
Hospital Rating Numeric,,,
1.0,1.045348,1.039733,169
2.0,1.023037,1.021250,567
3.0,1.005613,1.004250,840
4.0,0.985902,0.985642,758
5.0,0.964201,0.970300,287


Interpretation: Hospital overall rating showed a clear inverse association with Excess Readmission Ratio. Lower-rated hospitals tended to have ERRs above the expected benchmark, while 4- and 5-star hospitals averaged below expected levels.

In [102]:
rating_condition = (
    df_hospital_analysis
    .dropna(subset=["Hospital Rating Numeric"])
    .groupby(
        ["Hospital Rating Numeric", "Measure Name"]
    )
    .agg(
        Avg_ERR=("Excess Readmission Ratio", "mean"),
        Hospitals=("Facility ID", "nunique")
    )
    .reset_index()
)

rating_condition_pivot = (
    rating_condition
    .pivot(
        index="Hospital Rating Numeric",
        columns="Measure Name",
        values="Avg_ERR"
    )
)

rating_condition_pivot

Measure Name,READM-30-AMI-HRRP,READM-30-CABG-HRRP,READM-30-COPD-HRRP,READM-30-HF-HRRP,READM-30-HIP-KNEE-HRRP,READM-30-PN-HRRP
Hospital Rating Numeric,,,,,,
1.0,1.036211,1.075017,1.030815,1.047444,1.108379,1.042869
2.0,1.023273,1.029588,1.013652,1.024802,1.046283,1.024099
3.0,1.002729,1.014415,1.001665,1.005445,1.025586,1.000581
4.0,0.987013,0.979487,0.991183,0.983424,0.978643,0.987879
5.0,0.972402,0.950501,0.983356,0.961777,0.944577,0.972314


Interpretation: Higher hospital ratings were consistently associated with lower Excess Readmission Ratios across all six HRRP conditions. One-star hospitals averaged above the expected benchmark for every condition, while five-star hospitals averaged below expected for every condition. The relationship was therefore broad-based rather than concentrated in a single clinical measure.

# **Create Verified Metrics**

In [116]:
verified_findings = {
    "overall_condition": {
        "highest_avg_ERR_condition": "HIP-KNEE",
        "avg_ERR": 1.003985
    },

    "state_level": {
        "highest_avg_ERR_state": "MA",
        "avg_ERR": 1.034428,
        "avg_predicted_rate": 14.908686,
        "avg_expected_rate": 14.383138,
        "reporting_hospitals": 52
    },

    "massachusetts_AMI": {
        "avg_ERR": 1.051164,
        "median_ERR": 1.0285,
        "hospitals_with_reported_ERR": 33,
        "hospitals_above_expected": 23,
        "percent_above_expected": 69.7,
        "max_ERR": 1.2639
    },

    "ownership": {
        "proprietary_avg_ERR": 1.012651,
        "proprietary_median_ERR": 1.011725,
        "proprietary_hospitals": 544,
        "pattern": "Proprietary hospitals had average ERR above 1 across all six HRRP conditions",
        "condition_specific_hospital_counts": {
        "AMI": 306,
        "CABG": 136,
        "COPD": 407,
        "HF": 474,
        "HIP_KNEE": 268,
        "PN": 502
    }
    },

    "hospital_rating": {
        "1_star_avg_ERR": 1.045348,
        "2_star_avg_ERR": 1.023037,
        "3_star_avg_ERR": 1.005613,
        "4_star_avg_ERR": 0.985902,
        "5_star_avg_ERR": 0.964201,
        "1_star_hospitals": 169,
        "2_star_hospitals": 567,
        "3_star_hospitals": 840,
        "4_star_hospitals": 758,
        "5_star_hospitals": 287,
        "pattern": "Average ERR decreased consistently as hospital rating increased"
    },

    "rating_by_condition": {
        "1_star_all_conditions_above_expected": True,
        "5_star_all_conditions_below_expected": True,
        "strongest_gap_condition": "HIP-KNEE",
        "hip_knee_1_star_ERR": 1.108379,
        "hip_knee_5_star_ERR": 0.944577
    }
}

In [117]:
import json

with open("/content/verified_metrics_v2.json", "w") as f:
    json.dump(verified_findings, f, indent=2)

In [118]:
from google.colab import files
files.download("/content/verified_metrics_v2.json")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# **Build AI Prompt**

In [119]:
metrics_text = json.dumps(
    verified_findings,
    indent=2
)

In [120]:
prompt_v2 = f"""
You are assisting a healthcare operations analyst.

Use ONLY the verified metrics provided below.

Do not invent causes, clinical explanations, patient characteristics,
hospital attributes, or additional statistics.

Separate your response into:

1. Key observed findings
   - State only facts directly supported by the provided metrics.
   - Where ERR is above or below 1, translate it into an approximate
     percentage difference from expected when mathematically valid.

2. Analytical interpretation
   - Explain what the observed patterns may indicate.
   - Clearly distinguish interpretation from fact.

3. What cannot be concluded
   - Identify limitations, missing context, and sample-size constraints.
   - Do not make claims about hospital quality or causality.

4. Recommended follow-up analyses
   - Recommend only analyses that logically follow from the available data.
   - Avoid recommending analyses already completed unless a deeper version
     is specifically needed.
  - Do not recommend inferential statistical tests or confidence intervals
  unless the supplied data and analytical objective support statistical
  inference. Prefer descriptive comparisons, distributions, sample sizes,
  and sensitivity analyses for population-level reporting datasets.

5. Executive summary
   - Maximum 3 sentences.
   - Written for hospital leadership.
   - Focus on the most decision-relevant findings.

Important:
- ERR > 1 means predicted readmissions are above the expected benchmark.
- ERR < 1 means predicted readmissions are below expected.
- Distinguish absolute readmission rates from performance relative to expected.
- Do not describe a hospital, state, ownership type, or rating group as poor performing.
- Clearly label hypotheses as hypotheses.
- Mention sample-size limitations where relevant.
- Treat ownership and hospital rating findings as associations, not causal effects.
- Conclusions must apply only to hospitals with reported data.

Verified metrics:

{metrics_text}
"""

In [121]:
print(prompt_v2)


You are assisting a healthcare operations analyst.

Use ONLY the verified metrics provided below.

Do not invent causes, clinical explanations, patient characteristics,
hospital attributes, or additional statistics.

Separate your response into:

1. Key observed findings
   - State only facts directly supported by the provided metrics.
   - Where ERR is above or below 1, translate it into an approximate
     percentage difference from expected when mathematically valid.

2. Analytical interpretation
   - Explain what the observed patterns may indicate.
   - Clearly distinguish interpretation from fact.

3. What cannot be concluded
   - Identify limitations, missing context, and sample-size constraints.
   - Do not make claims about hospital quality or causality.

4. Recommended follow-up analyses
   - Recommend only analyses that logically follow from the available data.
   - Avoid recommending analyses already completed unless a deeper version
     is specifically needed.
  - Do not 

# **LLM API Integration**

In [122]:
from google.colab import userdata

api_key = userdata.get("OPENAI_API_KEY")

In [123]:
from openai import OpenAI

client = OpenAI(api_key=api_key)

# **AI Generated Insights**

In [124]:
response_v2 = client.responses.create(
    model="gpt-5.6-luna",
    input=prompt_v2
)

ai_response_v2 = response_v2.output_text

In [125]:
print(ai_response_v2)

## 1. Key observed findings

- **Overall condition:** HIP-KNEE had the highest average ERR among the conditions reported, at **1.003985**, or approximately **0.4% above expected**.
- **State level:** Massachusetts had the highest average ERR, at **1.034428**, approximately **3.4% above expected**. Its average predicted rate was **14.908686**, compared with an average expected rate of **14.383138**, a difference of **0.525548 rate units**. This summary included **52 reporting hospitals**.
- **Massachusetts AMI:** The average ERR was **1.051164**, approximately **5.1% above expected**; the median ERR was **1.0285**, approximately **2.9% above expected**. Of **33 hospitals** with reported AMI ERR, **23 hospitals**, or **69.7%**, were above expected. The maximum ERR was **1.2639**, approximately **26.4% above expected**.
- **Ownership:** Proprietary hospitals had an average ERR of **1.012651**, approximately **1.3% above expected**, with a median ERR of **1.011725**, approximately **1.2% a

In [126]:
with open("/content/ai_output_v2.txt", "w") as f:
    f.write(ai_response_v2)

files.download("/content/ai_output_v2.txt")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# **AI output Validation**


The AI-generated response was validated against the verified metrics calculated in Python before being accepted as the final project output.

Validation checks included:

- Confirming that all reported ERR values and percentages matched the Python-derived metrics.
- Verifying that ERR was interpreted correctly as performance relative to the expected benchmark.
- Confirming that absolute readmission rates were not confused with Excess Readmission Ratios.
- Checking that state-, ownership-, and hospital-rating conclusions applied only to hospitals with reported data.
- Confirming that the Massachusetts AMI finding was limited to the 33 hospitals with reported ERR values.
- Verifying that ownership findings were described as associations rather than causal effects.
- Verifying that the hospital-rating gradient was described as an association and not as evidence that higher ratings directly cause lower readmissions.
- Checking that the AI did not invent clinical explanations, patient characteristics, staffing issues, discharge practices, or other unsupported causes.
- Confirming that sample-size limitations were acknowledged, particularly for smaller ownership groups.
- Reviewing whether condition-specific ownership sample sizes were distinguished from the overall ownership sample.
- Comparing the API-generated output with the manually generated ChatGPT response using the same verified metrics and constrained prompt. The main findings and interpretations were consistent across both outputs.

The final AI response was therefore treated as an interpretation layer on top of verified analytical results rather than as an independent source of facts.

# **Key project Takaways**

- HIP-KNEE had the highest average Excess Readmission Ratio across the six HRRP conditions, although the average was only about 0.4% above the expected benchmark.

- Massachusetts had the highest average state-level ERR among the states analyzed, at approximately 3.4% above expected.

- Within Massachusetts AMI, approximately 69.7% of hospitals with reported ERR data were above their expected benchmark, indicating that the pattern was not driven solely by one or two hospitals.

- The analysis demonstrated that hospitals with higher absolute predicted readmission rates do not necessarily have higher Excess Readmission Ratios, because ERR evaluates performance relative to a risk-adjusted expected benchmark.

- Proprietary hospitals showed average ERR above 1.0 across all six HRRP conditions. The pattern was broad-based across conditions, although it should be interpreted as an association rather than a causal ownership effect.

- Hospital overall rating showed a clear inverse association with ERR. Average ERR declined from approximately 4.5% above expected for 1-star hospitals to approximately 3.6% below expected for 5-star hospitals.

- The hospital-rating pattern was consistent across all six HRRP conditions, suggesting that the overall relationship was not driven by a single clinical measure.

- Merging the HRRP dataset with CMS Hospital General Information enriched the analysis by allowing readmission performance to be compared across hospital ownership and rating categories.

- The GenAI workflow converted verified Python outputs into structured executive insights using a constrained prompt designed to reduce unsupported claims and hallucinations.

- Human validation remained essential. AI was used to summarize and interpret verified metrics, while Python remained the source of truth for all numerical findings.

# **AI Workflow Summary**

CMS Data
→ Data Cleaning and Validation
→ Python Analysis
→ Verified Metrics
→ Structured JSON
→ Constrained GenAI Prompt
→ GPT API Response
→ Human Validation
→ Executive Insights